assets/javascripts/discourse/components/events-calendar-subscription.js.es6

```diff
@@ -22,27 +22,31 @@ export default DropdownSelectBoxComponent.extend({
     ];
   },
 
-  @computed('userApiKey', 'category', 'siteSettings.login_required')
-  authSuffix(userApiKey, category, loginRequired) {
-    // only private sites need login
-    if (!loginRequired) return "";
-    // only append for private categories
-    if (!category || !category.read_restricted) return "";
+  @computed('userApiKey', 'siteSettings.login_required', 'category.read_restricted')
+  authSuffix(userApiKey, loginRequired, privateCategory) {
+    if (!loginRequired && !privateCategory) return "";
+    if (!userApiKey) return "";
+    let suffix = "";
     const {
       key,
       client_id,
     } = userApiKey;
     // only append if available
-    if (!key || !client_id) return "";
-    return `&user_api_key=${key}&user_api_client_id=${client_id}`;
+    if (key) {
+      suffix += `&user_api_key=${encodeURIComponent(key)}`;
+      if (client_id) {
+        // be aware that the client_id be changed at will by the user
+        suffix += `&user_api_client_id=${encodeURIComponent(client_id)}`;
```

Comment:
A user API key's client_id can be changed by simply passing a different one. It is optional but this is the effect that it has. I think the purpose of that feature is so that every client has at most one API key per user. Anyways, we need to make sure to encode it properly to prevent malicious third parties from injecting something weird into our URLs.